index.html

```diff
@@ -1929,11 +1929,17 @@ <h3>Add Proof Set/Chain</h3>
           </li>
           <li>
 Set |unsecuredDocument|.|proof| to |matchingProofs|.
-            <p class="note">
-This step is critical, as it <q>binds</q> the previous proofs to the document
+            <div class="note">
+                <p>
+This step means to add references to the graph names, as well as adding a copy 
+of <em>all</em> the claims contained in the proof graphs.
+                </p>
+                <p>
+The step is also critical, as it <q>binds</q> the previous proofs to the document
 prior to signing. The |proof| value for the document will be updated
 in a later step of this algorithm.
```

Comment:
```suggestion
The step is critical, as it <q>binds</q> any previous proofs to the document prior
to applying the current signature. The |proof| value for the document will be updated
in a later step of this algorithm.
```

Reply:
@TallTed (and @msporny @dlongley) I do not think this change is right, namely the 'the' -> 'any'. What the algorithm does is to take explicitly _the_ previous proof (or proofs, if the there are several), temporarily add that proof to the core data before hashing it. After the current proof is done, the previous proof _is removed_ from the dataset. Your change suggest that the current proof concerns the dataset and _any_ previous proofs, ie, it also concerns the "grandfather" proofs, which is not the case.

Reply:
@iherman,

> Your change suggest that the current proof concerns the dataset and any previous proofs, ie, it also concerns the "grandfather" proofs, which is not the case.

By virtue of including the hash of a previous proof (the "parent" proof in your parlance here, I believe), if that parent proof itself had a previous proof (a "grandparent" proof), the hash of the grandparent proof is transitively included.

As an example, for these proofs:

1. ProofA (no previous proof)
2. ProofB (ProofA is its previous proof)
3. ProofC (ProofB is its previous proof)

ProofC will include the hash of ProofB. But ProofB included the hash of ProofA. Therefore, ProofC includes the hash of ProofA.

Are we talking about something else though?

Reply:
> the 'the' -> 'any'

Allowing for both documents that carry some previous proof(s) and documents that carry *no* previous proof makes the language of this paragraph challenging.

"`the previous proofs`" strongly implies that this process *only* applies to documents that carry at least one, and probably more than one, previous proof.

"`any previous proofs`" implies (strongly, I think) that this process applies to documents that carry one, multiple, _or no_ previous proof(s).

Reply:
@dlongley no, we are talking about the same thing.

@TallTed: the algorithm, as described, concentrates on the creation of, say, ProofC (using @dlongley's formulation).  In ***that*** step, only the previous proof (ProofB) is involved, whatever the origin and the characteristics of that one are. For me, "any previous proof" suggests that, _in that step_, I have to take into account ProofA as well, which is not correct. I just want to avoid misunderstandings on the implementer's side.

Maybe adding into the note something along the lines of @dlongley's comment to make things more explicit is the right way to go. But I do not want to lie down the road over this; if you both are convinced that this is all right, be it.

(B.t.w., the current text does not come from me, it has been so for a long time; I have not really changed that sentence...)

Practicalities: I will be out of town for the week to come. @msporny, you can make the last commit and merge things if you prefer to close this while I am away.

Reply:
Possible alternative text:

"the matching proofs"

Reply:
"the previous proof, if present"?

Reply:
> For me, "any previous proof" suggests that, in that step, I have to take into account ProofA as well, which is not correct.

Referring to "the previous proofs" (note, plural!) suggests to me that both ProofA and ProofB must be considered when producing ProofC.

Referring to "the previous proof" (note, singular!) _could_ mean that only ProofB need be considered when producing ProofC. Some additional language tweaks might make this clearer.